File: modules/chat-core/src/main/java/com/shterneregen/securelan/chat/discovery/impl/DiscoveryMessageCodec.java
package com.shterneregen.securelan.chat.discovery.impl;

import com.shterneregen.securelan.chat.discovery.DiscoveryCodec;

final class DiscoveryMessageCodec {
    private DiscoveryMessageCodec() {
    }

    static String encode(DiscoveryMessage message) {
        return DiscoveryCodec.encode(new DiscoveryCodec.Message(
                message.peerId(),
                message.nickname(),
                message.chatPort(),
                message.filePort()
        ));
    }

    static DiscoveryMessage decode(String payload) {
        DiscoveryCodec.Message message = DiscoveryCodec.decode(payload);
        return new DiscoveryMessage(message.peerId(), message.nickname(), message.chatPort(), message.filePort());
    }

    static byte[] toBytes(DiscoveryMessage message) {
        return DiscoveryCodec.toBytes(new DiscoveryCodec.Message(
                message.peerId(),
                message.nickname(),
                message.chatPort(),
                message.filePort()
        ));
    }

    static DiscoveryMessage fromBytes(byte[] data, int length) {
        DiscoveryCodec.Message message = DiscoveryCodec.fromBytes(data, length);
        return new DiscoveryMessage(message.peerId(), message.nickname(), message.chatPort(), message.filePort());
    }
}

File: modules/chat-core/src/main/kotlin/com/shterneregen/securelan/chat/discovery/impl/DiscoveryMessageCodec.kt
package com.shterneregen.securelan.chat.discovery.impl

import com.shterneregen.securelan.chat.discovery.DiscoveryCodec

internal object DiscoveryMessageCodec {
    @JvmStatic
    fun encode(message: DiscoveryMessage): String = DiscoveryCodec.encode(toPublicMessage(message))

    @JvmStatic
    fun decode(payload: String?): DiscoveryMessage = fromPublicMessage(DiscoveryCodec.decode(payload))

    @JvmStatic
    fun toBytes(message: DiscoveryMessage): ByteArray = DiscoveryCodec.toBytes(toPublicMessage(message))

    @JvmStatic
    fun fromBytes(data: ByteArray, length: Int): DiscoveryMessage = fromPublicMessage(DiscoveryCodec.fromBytes(data, length))

    private fun toPublicMessage(message: DiscoveryMessage): DiscoveryCodec.Message =
        DiscoveryCodec.Message(message.peerId, message.nickname, message.chatPort, message.filePort)

    private fun fromPublicMessage(message: DiscoveryCodec.Message): DiscoveryMessage =
        DiscoveryMessage(message.peerId, message.nickname, message.chatPort, message.filePort)
}
